### DISCUSSION TITLE
Some Docker Issues

### DISCUSSION THREAD

**Discussion by ghost** [Woolverine94/biniou#12] [General]
I was having some issues getting Docker running with Cuda on Ubuntu. Not sure if that file is being maintained frequently but take a look what worked for me, feel free to commit. You'd need to run this as a replacement for the original Dockerfile, not as an upgrade.

# Start with an official Python base image
FROM python:3.11.6
# Avoid interactive prompts from apt (e.g., tzdata)
ENV DEBIAN_FRONTEND=noninteractive

# Install necessary system dependencies
RUN apt-get update && apt-get -y upgrade && \
    apt-get install -y bash sudo apt-utils git gcc perl make ffmpeg openssl curl

# Setup user biniou and use it to install
RUN adduser --disabled-password --gecos '' biniou
USER biniou
WORKDIR /home/biniou

# Pull repo
RUN git clone https://github.com/Woolverine94/biniou.git biniou
WORKDIR /home/biniou/biniou

# Create a virtual environment and install Python packages
RUN python -m venv env
SHELL ["/bin/bash", "-c"]
RUN source env/bin/activate && \
    pip install -U pip wheel && \
    pip install -U torch==2.1.0 torchvision torchaudio && \
    FORCE_CMAKE=1 pip install -U llama-cpp-python && \
    pip install -r requirements.txt

# Install biniou using the provided install script
RUN ./install.sh

# Set up directories and permissions
RUN mkdir -p /home/biniou/.cache/huggingface /home/biniou/biniou/gfpgan
RUN chmod +x /home/biniou/biniou/webui.sh

# Exposing port 7860
EXPOSE 7860/tcp

# Launch at startup
CMD [ "/home/biniou/biniou/webui.sh" ]

# Reset DEBIAN_FRONTEND for interactive use when running the container
ENV DEBIAN_FRONTEND=dialog

**Comment by Woolverine94** [comment]
Hello @AndrewDigitalChaos,

Thanks for your feedback. 

Default biniou installation is natively CPU-only (and it will stay that way), so it don't work out-of-the-box with CUDA in a docker env. 

Since commit [da3be41](https://github.com/Woolverine94/biniou/commit/da3be417598cfed1f90d896974ae1383c3d14b6c) of yesterday, you *should* be able to activate CUDA directly from biniou's webUI control panel, even in docker. 
Unfortunately, I don't have access to CUDA-compatible hardware to validate it's working, even if I can confirm that replacement of pyTorch flavor is done by the update script. 

Anyway, this is more a workaround for using CUDA in docker than a turnkey CUDA docker installation. 

Thanks A LOT for your contribution, I will think about a way later today to introduce your changes (probably via a specific dockerfile).

**Comment by Woolverine94** [comment]
@AndrewDigitalChaos,

Following your ideas, I've added a specific CUDA Dockerfile that can be used as following to build a docker image : 

```bash
docker build -t biniou https://raw.githubusercontent.com/Woolverine94/biniou/main/CUDA/Dockerfile
```

Thanks again for your contribution to biniou.
